# Practico Mentoria - Analisis Exploratorio y Curación de Datos

---

<h1><FONT COLOR="#e600e5"> Autor: Melania Omonte</FONT></h1>

### Importaciones

In [181]:
%matplotlib inline

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import scipy as sp

from sklearn import preprocessing

import warnings
warnings.filterwarnings('ignore')

In [150]:
# Seteamos una semilla para Reproducibilidad
np.random.seed(0)

<h3> Carga de los Datasets </h3>

In [151]:
player_df = pd.read_csv('./Datasets/football_player.csv')
team_df = pd.read_csv('./Datasets/football_team.csv')
match_df = pd.read_csv('./Datasets/football_match.csv')

<h2>Exploremos un poco los Datasets</h2>

<h3>Players Dataset</h3>

In [152]:
print("Shape 'player_df' = {}".format(player_df.shape))
player_df.sample(5)

Shape 'player_df' = (11060, 40)


player name    birthday  height_m  weight_kg  overall_rating  \
1534           Carlos Acuna  1988-06-23      1.78      71.21           67.33   
7238       Max Christiansen  1996-09-25      1.88      83.91           64.09   
10999        Zakaria M'Sila  1992-04-06      1.78      74.84           59.00   
2669   Dimitrija Lazarevski  1982-09-23      1.78      74.84           59.00   
1403                  Bruma  1994-10-24      1.73      66.22           75.26   

       potential preferred foot  crossing  finishing  heading accuracy  ...  \
1534       71.81          right     51.52      67.43             68.86  ...   
7238       74.45          right     47.73      37.73             60.82  ...   
10999      65.10           left     57.30      50.90             50.20  ...   
2669       61.00           left     51.00      38.00             54.00  ...   
1403       85.71          right     67.26      64.34             51.09  ...   

       vision  penalties  marking  standing tackle  sliding tackle  gk_diving  \
1534    44.00      66.67    25.52            30.05           29.00      15.24   
7238    56.45      47.00    53.36            67.00           62.36      14.73   
10999   54.30      54.80    27.50            32.70           23.30      12.30   
2669      NaN      61.00    60.00            61.00             NaN       5.00   
1403    66.89      70.94    20.71            26.29           22.20      10.09   

       gk_handling  gk_kicking  gk_positioning  gk_reflexes  
1534         12.43       16.43           12.52        11.76  
7238          6.73       10.73           13.73         8.73  
10999        15.30       10.30            7.30        13.30  
2669         22.00       45.00           22.00        22.00  
1403         12.09        7.09           11.09         7.09  

[5 rows x 40 columns]

<h3>Team Dataset</h3>

In [153]:
print("Shape 'team_df'   = {}".format(team_df.shape))
team_df.sample(5)

Shape 'team_df'   = (288, 22)


team long name team short name  buildUpPlaySpeed  \
185               Lechia Gdańsk             LGD             50.83   
222                 FC Penafiel             PEN             54.00   
198              Pogoń Szczecin             POG             55.67   
197  Podbeskidzie Bielsko-Biała             POD             62.00   
173                   Excelsior             EXC             57.67   

    buildUpPlaySpeedClass buildUpPlayDribblingClass  buildUpPlayPassing  \
185              Balanced                    Little               48.33   
222              Balanced                    Normal               39.00   
198              Balanced                    Little               42.00   
197              Balanced                    Little               58.50   
173              Balanced                    Little               60.00   

    buildUpPlayPassingClass buildUpPlayPositioningClass  \
185                   Mixed                   Organised   
222                   Mixed                   Organised   
198                   Mixed                   Organised   
197                   Mixed                   Organised   
173                   Mixed                   Organised   

     chanceCreationPassing chanceCreationPassingClass  ...  \
185                  54.83                     Normal  ...   
222                  52.00                     Normal  ...   
198                  53.33                     Normal  ...   
197                  57.00                     Normal  ...   
173                  62.67                     Normal  ...   

     chanceCreationShooting chanceCreationShootingClass  \
185                   61.33                        Lots   
222                   38.00                      Normal   
198                   57.00                      Normal   
197                   45.50                      Normal   
173                   44.00                      Normal   

     chanceCreationPositioningClass defencePressure defencePressureClass  \
185                       Organised           50.33               Medium   
222                       Organised           40.00               Medium   
198                       Organised           51.33               Medium   
197                       Organised           38.00                 Deep   
173                       Organised           25.67                 Deep   

     defenceAggression defenceAggressionClass  defenceTeamWidth  \
185              58.83                  Press             56.67   
222              35.00                  Press             46.00   
198              53.67                  Press             52.33   
197              46.00                  Press             43.00   
173              39.33                  Press             33.67   

    defenceTeamWidthClass  defenceDefenderLineClass  
185                Normal                     Cover  
222                Normal                     Cover  
198                Normal                     Cover  
197                Normal                     Cover  
173                Narrow                     Cover  

[5 rows x 22 columns]

<h3>Match Dataset</h3>

In [154]:
print("Shape 'match_df'  = {}".format(match_df.shape))
match_df.sample(5)

Shape 'match_df'  = (25979, 12)


country name             league name     season  stage  \
12042        Italy           Italy Serie A  2012/2013     35   
8575       Germany   Germany 1. Bundesliga  2010/2011     25   
9067       Germany   Germany 1. Bundesliga  2012/2013     12   
13165        Italy           Italy Serie A  2015/2016     34   
14904  Netherlands  Netherlands Eredivisie  2013/2014      2   

                      date  home team long name home short long name  \
12042  2013-05-05 00:00:00              Catania                  CAT   
8575   2011-03-05 00:00:00        VfB Stuttgart                  STU   
9067   2012-11-17 00:00:00  Eintracht Frankfurt                  EFR   
13165  2016-04-21 00:00:00                Milan                  ACM   
14904  2013-08-10 00:00:00      Heracles Almelo                  HER   

      away team long name away short long name  home team goal  \
12042               Siena                  SIE               3   
8575        FC Schalke 04                  S04               1   
9067          FC Augsburg                  AUG               4   
13165               Carpi                  CAP               0   
14904          PEC Zwolle                  ZWO               1   

       away team goal  total goal  
12042               0           3  
8575                0           1  
9067                2           6  
13165               0           0  
14904               3           4

<h2>Exploremos un poco los Datasets y sus correspondientes Tipos</h2>

<h3>Players Dtypes</h3>


In [155]:
player_df.dtypes

player name            object
birthday               object
height_m              float64
weight_kg             float64
overall_rating        float64
potential             float64
preferred foot         object
crossing              float64
finishing             float64
heading accuracy      float64
short passing         float64
volleys               float64
dribbling             float64
curve                 float64
free kick accuracy    float64
long passing          float64
ball control          float64
acceleration          float64
sprint speed          float64
agility               float64
reactions             float64
balance               float64
shot power            float64
jumping               float64
stamina               float64
strength              float64
long shots            float64
aggression            float64
interceptions         float64
positioning           float64
vision                float64
penalties             float64
marking               float64
standing t

<h3>Match Dtypes</h3>

In [156]:
match_df.dtypes

country name            object
league name             object
season                  object
stage                    int64
date                    object
home team long name     object
home short long name    object
away team long name     object
away short long name    object
home team goal           int64
away team goal           int64
total goal               int64
dtype: object

<h3>Team Dtypes</h3>

In [157]:
team_df.dtypes

team long name                     object
team short name                    object
buildUpPlaySpeed                  float64
buildUpPlaySpeedClass              object
buildUpPlayDribblingClass          object
buildUpPlayPassing                float64
buildUpPlayPassingClass            object
buildUpPlayPositioningClass        object
chanceCreationPassing             float64
chanceCreationPassingClass         object
chanceCreationCrossing            float64
chanceCreationCrossingClass        object
chanceCreationShooting            float64
chanceCreationShootingClass        object
chanceCreationPositioningClass     object
defencePressure                   float64
defencePressureClass               object
defenceAggression                 float64
defenceAggressionClass             object
defenceTeamWidth                  float64
defenceTeamWidthClass              object
defenceDefenderLineClass           object
dtype: object

---

# 1. Importacion de los datos

<h2>Calculemos el rango de fechas de los partidos</h2>

Antes de calcular el rango de fechas de los partidos, debemos validar que tipo de objeto es la fecha

In [158]:
match_df.dtypes['date']

dtype('O')

Como la fecha es un campo del tipo object, no podremos calcular el rango solicitado,
por lo tanto tendremos que cambiar el tipo, asi podemos generar el valor solicitado.

<h3>Modificamos el tipo "date", para poder calcular el rango</h3>

In [159]:

match_df2 = pd.read_csv("./Datasets/football_match.csv", parse_dates=["date"])



<h3>Visualizamos que haya cambiado el tipo "date"</h3>

In [160]:

match_df2.dtypes['date']


dtype('<M8[ns]')

Validamos que se cambio el tipo a datetime64[ns]

<h3>Realizamos la diferencia, para poder calcular el rango solicitado</h3>

In [161]:
match_df2['date'].max() - match_df2['date'].min()

Timedelta('2868 days 00:00:00')

<h3><FONT COLOR="#008f39">Rta: El rango de fechas entre partidos es 2868 dias.</FONT></h3>

# 2. Etiquetas de variables/columnas: no usar caracteres especiales

Chequar que no haya caracteres fuera de `a-Z`, `0-9` y `_` en los nombres de columnas de los Dataframes:
* `player_df`
* `team_df`
* `match_df`

<h2>Exploramos los Datasets y validamos que no hayan caracteres fuera de lo solicitado</h2>

<h3>Match DataSet</h3>

In [162]:
match_df.columns[~match_df.columns.str.match(r'^(\w+)$')]

Index(['country name', 'league name', 'home team long name',
       'home short long name', 'away team long name', 'away short long name',
       'home team goal', 'away team goal', 'total goal'],
      dtype='object')

Chequeamos que existen varias columnas que tienen caracteres fuera de "a-Z, 0-9 y _" en el dataset match.

<h3>Player DataSet</h3>

In [163]:
player_df.columns[~player_df.columns.str.match(r'^(\w+)$')]

Index(['player name', 'preferred foot', 'heading accuracy', 'short passing',
       'free kick accuracy', 'long passing', 'ball control', 'sprint speed',
       'shot power', 'long shots', 'standing tackle', 'sliding tackle'],
      dtype='object')

Chequeamos que existen varias columnas que tienen caracteres fuera de "a-Z, 0-9 y _" en el dataset player.

<h3>Team DataSet</h3>

In [164]:
team_df.columns[~team_df.columns.str.match(r'^(\w+)$')]

Index(['team long name', 'team short name'], dtype='object')

Chequeamos que existen 2 columnas que tienen caracteres fuera de "a-Z, 0-9 y _" en el dataset team.

<h3>Reemplazamos los valores fuera de "a-Z, 0-9 y _" en el dataset team</h3>

In [165]:
team_df.columns = team_df.columns.str.replace(' ', '_')
team_df.head()

team_long_name team_short_name  buildUpPlaySpeed buildUpPlaySpeedClass  \
0           KRC Genk             GEN             56.33              Balanced   
1       Beerschot AC             BAC             46.00              Balanced   
2   SV Zulte-Waregem             ZUL             55.50              Balanced   
3   Sporting Lokeren             LOK             64.00              Balanced   
4  KSV Cercle Brugge             CEB             53.67              Balanced   

  buildUpPlayDribblingClass  buildUpPlayPassing buildUpPlayPassingClass  \
0                    Little               44.33                   Mixed   
1                    Little               41.50                   Mixed   
2                    Little               52.67                   Mixed   
3                    Little               53.50                   Mixed   
4                    Little               44.17                   Mixed   

  buildUpPlayPositioningClass  chanceCreationPassing  \
0                   Organised                  55.67   
1                   Organised                  46.00   
2                   Organised                  50.00   
3                   Organised                  44.00   
4                   Organised                  59.00   

  chanceCreationPassingClass  ...  chanceCreationShooting  \
0                     Normal  ...                   55.83   
1                     Normal  ...                   53.75   
2                     Normal  ...                   41.50   
3                     Normal  ...                   54.33   
4                     Normal  ...                   48.00   

  chanceCreationShootingClass  chanceCreationPositioningClass defencePressure  \
0                      Normal                       Organised           49.17   
1                      Normal                       Organised           48.25   
2                      Little                       Organised           48.67   
3                      Normal                       Organised           48.83   
4                      Normal                       Organised           60.17   

  defencePressureClass  defenceAggression defenceAggressionClass  \
0               Medium              50.67                  Press   
1               Medium              49.00                  Press   
2               Medium              56.17                  Press   
3               Medium              55.17                  Press   
4               Medium              49.67                  Press   

   defenceTeamWidth defenceTeamWidthClass  defenceDefenderLineClass  
0             59.83                Normal                     Cover  
1             58.75                Normal                     Cover  
2             47.83                Normal                     Cover  
3             55.50                Normal                     Cover  
4             64.67                Normal                     Cover  

[5 rows x 22 columns]

<h3>Validamos que se hayan reemplazado bien los campos en el dataset Team</h3>

In [166]:
team_df.columns[~team_df.columns.str.match(r'^(\w+)$')]

Index([], dtype='object')

Validamos que se reemplazaron exitosamente los campos, ya que la consulta anterior no nos devuelve ningun campo.

<h3>Reemplazamos los valores fuera de "a-Z, 0-9 y _" en el dataset Player</h3>

In [167]:
player_df.columns = player_df.columns.str.replace(' ', '_')
player_df.head()

player_name    birthday  height_m  weight_kg  overall_rating  \
0  Aaron Appindangoye  1992-02-29      1.83      84.82           63.60   
1     Aaron Cresswell  1989-12-15      1.70      66.22           66.97   
2         Aaron Doran  1991-05-13      1.70      73.94           67.00   
3       Aaron Galindo  1982-05-08      1.83      89.81           69.09   
4        Aaron Hughes  1979-11-08      1.83      69.85           73.24   

   potential preferred_foot  crossing  finishing  heading_accuracy  ...  \
0      67.60          right     48.60      43.60             70.60  ...   
1      74.48           left     70.79      49.45             52.94  ...   
2      74.19          right     68.12      57.92             58.69  ...   
3      70.78          right     57.22      26.26             69.26  ...   
4      74.68          right     45.08      38.84             73.04  ...   

   vision  penalties  marking  standing_tackle  sliding_tackle  gk_diving  \
0   53.60      47.60    63.80            66.00           67.80       5.60   
1   57.45      53.12    69.39            68.79           71.52      12.18   
2   69.38      60.54    22.04            21.12           21.35      14.04   
3   53.78      41.74    70.61            70.65           68.04      14.17   
4   46.48      52.96    77.60            76.04           74.60       8.28   

   gk_handling  gk_kicking  gk_positioning  gk_reflexes  
0        10.60        9.60            7.60         7.60  
1         8.67       14.24           10.36        12.91  
2        11.81       17.73           10.12        13.50  
3        11.17       22.87           11.17        10.17  
4         8.32       24.92           12.84        11.92  

[5 rows x 40 columns]

<h3>Validamos que se hayan reemplazado bien los campos en el dataset Player</h3>

In [168]:
player_df.columns[~player_df.columns.str.match(r'^(\w+)$')]

Index([], dtype='object')

Validamos que se reemplazaron exitosamente los campos, ya que la consulta anterior no nos devuelve ningun campo.

<h3>Reemplazamos los valores fuera de "a-Z, 0-9 y _" en el dataset Match</h3>

In [169]:
match_df.columns = match_df.columns.str.replace(' ', '_')
match_df.head()

country_name             league_name     season  stage                 date  \
0      Belgium  Belgium Jupiler League  2008/2009      1  2008-08-17 00:00:00   
1      Belgium  Belgium Jupiler League  2008/2009      1  2008-08-16 00:00:00   
2      Belgium  Belgium Jupiler League  2008/2009      1  2008-08-16 00:00:00   
3      Belgium  Belgium Jupiler League  2008/2009      1  2008-08-17 00:00:00   
4      Belgium  Belgium Jupiler League  2008/2009      1  2008-08-16 00:00:00   

  home_team_long_name home_short_long_name away_team_long_name  \
0            KRC Genk                  GEN        Beerschot AC   
1    SV Zulte-Waregem                  ZUL    Sporting Lokeren   
2   KSV Cercle Brugge                  CEB      RSC Anderlecht   
3            KAA Gent                  GEN           RAEC Mons   
4       FCV Dender EH                  DEN   Standard de Liège   

  away_short_long_name  home_team_goal  away_team_goal  total_goal  
0                  BAC               1               1           2  
1                  LOK               0               0           0  
2                  AND               0               3           3  
3                  MON               5               0           5  
4                  STL               1               3           4

<h3>Validamos que se hayan reemplazado bien los campos en el dataset Match</h3>

In [170]:
match_df.columns[~match_df.columns.str.match(r'^(\w+)$')]

Index([], dtype='object')

Validamos que se reemplazaron exitosamente los campos, ya que la consulta anterior no nos devuelve ningun campo.

# 3. Agregar nuevas caracteristicas

Agregar al Dataframe `player_df` una nueva columna que sea `imc` correspondiente al **Indice de Masa Corporal**

Link:
* https://www.texasheart.org/heart-health/heart-information-center/topics/calculadora-del-indice-de-masa-corporal-imc/

In [171]:
from sklearn import preprocessing
player_df_mod = pd.read_csv('./Datasets/football_player.csv')
player_df_mod.head()

player name    birthday  height_m  weight_kg  overall_rating  \
0  Aaron Appindangoye  1992-02-29      1.83      84.82           63.60   
1     Aaron Cresswell  1989-12-15      1.70      66.22           66.97   
2         Aaron Doran  1991-05-13      1.70      73.94           67.00   
3       Aaron Galindo  1982-05-08      1.83      89.81           69.09   
4        Aaron Hughes  1979-11-08      1.83      69.85           73.24   

   potential preferred foot  crossing  finishing  heading accuracy  ...  \
0      67.60          right     48.60      43.60             70.60  ...   
1      74.48           left     70.79      49.45             52.94  ...   
2      74.19          right     68.12      57.92             58.69  ...   
3      70.78          right     57.22      26.26             69.26  ...   
4      74.68          right     45.08      38.84             73.04  ...   

   vision  penalties  marking  standing tackle  sliding tackle  gk_diving  \
0   53.60      47.60    63.80            66.00           67.80       5.60   
1   57.45      53.12    69.39            68.79           71.52      12.18   
2   69.38      60.54    22.04            21.12           21.35      14.04   
3   53.78      41.74    70.61            70.65           68.04      14.17   
4   46.48      52.96    77.60            76.04           74.60       8.28   

   gk_handling  gk_kicking  gk_positioning  gk_reflexes  
0        10.60        9.60            7.60         7.60  
1         8.67       14.24           10.36        12.91  
2        11.81       17.73           10.12        13.50  
3        11.17       22.87           11.17        10.17  
4         8.32       24.92           12.84        11.92  

[5 rows x 40 columns]

<h3>Saco el Cuadrado de la altura, para poder sacar el IMC.
Una vez definido el cuadrado, calculo el IMC</h3>

In [172]:
altura_cuadrado=player_df_mod['height_m']**2
def imc(peso,altura):
  #  return peso / (altura*altura)
    return peso / (altura_cuadrado)

<h3> Agrego la columna IMC en el dataFrame Player</h3>

In [173]:

player_df_mod['IMC'] = player_df_mod.apply(lambda x: imc(x.weight_kg, x.height_m), axis=1)
player_df_mod.describe()

ValueError: Wrong number of items passed 11060, placement implies 1

<h2> Grafico de distribucion de IMC</h2>

In [ ]:

plt.figure(figsize=(10,6))

## Grafico la distribucion de Masa Corporal"  
sns.distplot(player_df_mod.IMC, kde=True, bins=5, label='IMC')

plt.ylabel('Densidad de Masa Corporal')
plt.legend()

<h2> Visualizamos los valores atipicos para el calculo realizado de IMC</h2>

In [ ]:
#Para observar valores atípicos visualizamos el gráfico de caja...
plt.figure(figsize=(10,6))
data = player_df_mod[['IMC']]
sns.boxplot(data = data, orient="h", palette="coolwarm")
#sns.stripplot(data=data, color='black')
plt.show()

# 4. Tratar valores faltantes

Veamos cuantos valores nulos tenemos

In [ ]:
player_missing_values_count = player_df.isnull().sum()

player_missing_values_count[player_missing_values_count > 0]

Tenemos 478 valores nulos en 7 columnas del DataFrame Player

In [ ]:
len( player_df.dropna())/len(player_df)

In [ ]:
len(player_df.dropna(subset=['volleys']))/len(player_df)

In [ ]:
player_df[player_df.volleys.isnull()]

<h2>Eliminamos los valores nulos</h2>

In [ ]:
player_df_mod = player_df_mod.dropna(subset=['volleys'])

In [ ]:
player_df_mod.describe()

<h2>Validamos que se hayan validado esos valores nulos</h2>

In [ ]:

player_missing_values_count = player_df_mod.isnull().sum()

player_missing_values_count[player_missing_values_count > 0]

Validamos que se eliminaron exitosamente los campos nulos, ya que no devuelve ningun valor la consulta realizada.

In [ ]:
player_df[player_df.volleys.isnull()]

Algunas tecnicas para tratar los _missing values_:
* **Eliminar** muestras o variables que tienen datos faltantes.
* **Imputar** los valores perdidos, es decir, sustituirlos por estimaciones por ejemplo la `media`, la `moda` ó usando `KNN`.

A) Analizar si es conveniente **Eliminar** las muestras o variables con datos faltantes del Dataframe `player_df`. 

B) Aplicar la **Imputacion** usando la `media` o `moda` sobre las columnas con _missing values_ del Dataframe `player_df`.

**Hint**:
* Para la imputacion usando la `media`, `moda` ver el siguiente link:  
    https://pandas.pydata.org/pandas-docs/stable/user_guide/missing_data.html

### ¿Eliminar los _missing values_? Justificar

# Elimino los valores en un dataset modificado
Para eliminar los valores missing, debemos realizar un analisis sobre esos campos.Las columnas con valores missing
son:
volleys           478
curve             478
agility           478
balance           478
jumping           478
vision            478
sliding_tackle    478

Una de las soluciones para resolver este problema podria ser llenar estos los valores con ceros. Pero esta solucion
no seria la mas optima, porque por ejemplo para un jugador, tendriamos que su agilidad es cero, y no seria representativo.

Otra de las opciones para resolver este problema es decidir eliminar estos valores, suponiendo que los valores 
missing pueden ser errores, de esta manera subsanamos dicho error.

Otra de las opciones es la imputacion de la Media o Moda, segun corresponda, analisis detallado mas abajo.


<h2> Imputacion usando Media y Moda</h2>
<h3> Reemplazo de Valores Faltantes usando la moda</h3>

In [ ]:
# Rellenamos usando la Moda
# player_df.fillna(player_df.mode(), inplace=True)

player_df_reemplazo_nan_moda = player_df

for column in ['volleys','agility','curve','balance','jumping','vision','sliding_tackle']:
    player_df_reemplazo_nan_moda[column].fillna(player_df_reemplazo_nan_moda[column].mode()[0], inplace=True)

player_df_reemplazo_nan_moda



<h3>Validamos que se hayan reemplazado bien los valores y que no hayan valores missing:</h3>

In [ ]:
player_missing_values_count = player_df.isnull().sum()

player_missing_values_count[player_missing_values_count > 0]

Se comprueba exitosamente que no hay valores missing, una vez que se reemplazaron los datos por su moda.

<h3> Reemplazo de Valores Faltantes usando la Media</h3>

In [ ]:
# Rellenamos usando la Moda
player_df_reemplazo_nan_media = player_df
player_df_reemplazo_nan_media.fillna(player_df_reemplazo_nan_media.mean(), inplace=True)


In [174]:
player_missing_values_count = player_df_reemplazo_nan_media.isnull().sum()

player_missing_values_count[player_missing_values_count > 0]

Series([], dtype: int64)

Se comprueba exitosamente que no hay valores missing, una vez que se reemplazaron los datos por su media.

# 5. Normalizacion de columnas


Normalizar la columna `crossing` usando **Min-Max**.

Normalizar la columna `short_passing` usando **Z-score**.


<h2> Normalizando la columna crossing, usando Min-Max</h2>

Normalizamos la columna y mostramos un listado antes de la normalizacion y despues de la misma

In [175]:
# TODO
print(player_df.crossing.head(10))

scaler = preprocessing.MinMaxScaler()
player_df[["crossing"]] = scaler.fit_transform(player_df[["crossing"]])

print(player_df.crossing.head(10))

0    48.60
1    70.79
2    68.12
3    57.22
4    45.08
5    73.89
6    47.57
7    78.04
8    12.00
9    63.89
Name: crossing, dtype: float64
0    0.511036
1    0.777231
2    0.745202
3    0.614443
4    0.468810
5    0.814419
6    0.498680
7    0.864203
8    0.071977
9    0.694458
Name: crossing, dtype: float64


<h2>Normalizamos la columna short_passing usando Z-score</h2>

Normalizamos la columna y mostramos un listado antes de la normalizacion y despues de la misma.

In [176]:

print(player_df["short_passing"].head(10))

scaler = preprocessing.MinMaxScaler()
player_df[["short_passing"]] = sp.stats.zscore(player_df[["short_passing"]])

print(player_df["short_passing"].head(10))

0    60.60
1    62.27
2    65.12
3    64.70
4    64.76
5    78.26
6    63.57
7    76.27
8    23.00
9    68.95
Name: short_passing, dtype: float64
0    0.017238
1    0.140868
2    0.351853
3    0.320761
4    0.325202
5    1.324605
6    0.237107
7    1.177285
8   -2.766282
9    0.635387
Name: short_passing, dtype: float64


# 6. Codificar variables

> Las variables categóricas deben ser etiquetadas como variables numéricas, no como cadenas.

Codificar la variable `country_name` del Dataframe `match_df`

<h2> La columna "country_name" antes de ser etiquetada como variable numerica</h2>

In [177]:

print(set(match_df["country_name"]))

{'Portugal', 'Belgium', 'Netherlands', 'France', 'England', 'Scotland', 'Italy', 'Poland', 'Switzerland', 'Spain', 'Germany'}


<h2>Etiqueto como variables numericas a la columna "country_name"</h2>

In [178]:
le = preprocessing.LabelEncoder()
match_df[["country_name"]] = le.fit_transform(match_df[["country_name"]])

<h2>Visualizo la columna, con los datos ya transformados</h2>

In [179]:
print(set(match_df["country_name"]))


{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10}


A la columna "country_name", la etiquedamos con variables numericas